pkg/kubelet/kubelet_pods.go

```diff
@@ -1779,21 +1778,17 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 			continue
 		}
 
-		if pod.DeletionTimestamp == nil {
-			continue
-		}
-
-		// and if the pod itself is being deleted, then the CRI may have removed the container already and for whatever reason the kubelet missed the exit code
-		// (this seems not awesome).  We know at this point that we will not be restarting the container.
```

Comment:
There's no reason we should skip pods that were not deleted from what I can see. This is the cause of the bug.

Reply:
nice catch, I agree this is the root cause
/lgtm